packages/room/channel/channel.js

```diff
@@ -120,28 +121,58 @@ export const createChannel = ({ api, event, peer, streams }) => {
     }
 
     _onError = async () => {
-      const errorTime = Date.now()
-
-      if (this._roomId && this._clientId) {
-        const response = await this._api.getClient(this._roomId, this._clientId)
-
-        if (response.code === 404) {
-          this.disconnect()
-          this._event.emit(RoomEvent.CHANNEL_CLOSED, {
-            reason: REASONS.NOT_FOUND,
-          })
-          return
-        }
-
-        // Reconnect
-        if (errorTime - this._startTime < 1000) {
+      const onError = async () => {
+        if (!navigator.onLine) {
           setTimeout(() => {
-            this._reconnect()
+            onError()
```

Comment:
Will `onError()` keep trying to call itself when the user is offline?

Reply:
it will keep trying call onError() while the user is still offline in every one second, once the connection is gained it will check the client existence and stops the polling

should we add a 5 seconds timeout to it, if it's has been out of connection for that given time it will automatically emit the `Channel_Closed` event ? with `TIMEOUT` as the reason ? since the Hub api automatically kick clients that has not been connected to the SSE for more than 5 seconds

Reply:
I think 1 second is fine because onError only process the part of checking the `navigator.onLine` before it is recalled again. When the user is online it will not be called again until channel error again.

Reply:
When the client is kicked, the reason will be `NOT_FOUND`. 

Or should we just change `NOT_FOUND` into `TIMEOUT` instead? I think `TIMEOUT` reason is better naming because when the client is kicked, the connection will be lost won't it?  And, when the connection is lost, there will be timeout until 5 seconds. The connection will be `TIMEOUT` and should be closed.

Reply:
ah what i mean about the 5 second timeout is not to change the polling rate from 1 second to 5, but add additional checks, to emit the Channel_Closed - TIMEOUT if the code has been polling the `onError()` function for more than 5 second

Reply:
I see, that's fine for me